eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -0,0 +1,258 @@
+      $pkgKey = $pkg.Package
+      if (!$packageLookup.ContainsKey($pkgKey)) {
+        $packageLookup[$pkgKey] = $pkg
+      }
+      else {
+        $packageValue = $packageLookup[$pkgKey]
+        Write-Host "Found more than one package entry for $($packageValue.Package) selecting the first one with groupId $($packageValue.GroupId), skipping $($pkg.GroupId)"
+      }
+    }
+  }
+
+  return $packageLookup
+}
+
+$onboardedPackages = &$GetOnboardedDocsMsPackagesFn `
+  -DocRepoLocation $DocRepoLocation
+
+# This criteria is different from criteria used in `Update-DocsMsPackages.ps1`
+# because we need to generate ToCs for packages which are not necessarily "New"
+# in the metadata AND onboard legacy packages (which `Update-DocsMsPackages.ps1`
+# does not do)
+$metadata = (Get-CSVMetadata).Where({
+    $_.Package `
+      -and $onboardedPackages.ContainsKey($_.Package) `
+      -and $_.Hide -ne 'true'
+  })
+
+$fileMetadata = @()
+foreach ($metadataFile in Get-ChildItem "$DocRepoLocation/metadata/*/*.json" -Recurse) {
+  $fileContent = Get-Content $metadataFile -Raw
+  $metadataEntry = ConvertFrom-Json $fileContent
+
+  if ($metadataEntry) {
+    $fileMetadata += $metadataEntry
+  }
+}
+
+# Add file metadata information to package metadata from metadata CSV. Because
+# metadata can exist for packages in both preview and GA there may be more than
+# one file metadata entry. If that is the case keep the first entry found. We
+# only use the `DirectoryPath` property from the json file metadata at this time
+for ($i = 0; $i -lt $metadata.Count; $i++) {
+  foreach ($fileEntry in $fileMetadata) {
+    if ($fileEntry.Name -eq $metadata[$i].Package) {
+      if ($metadata[$i].PSObject.Members.Name -contains "FileMetadata") {
+        Write-Host "File metadata already added for $($metadata[$i].Package). Keeping the first entry found."
+    continue
+  }
+
+      Add-Member `
+        -InputObject $metadata[$i] `
+        -MemberType NoteProperty `
+        -Name FileMetadata `
+        -Value $fileEntry
+  }
+  }
+}
+
+$packagesForToc = @{}
+foreach ($metadataEntry in (GetPackageLookup $metadata).Values) {
+  if (!$metadataEntry.ServiceName) {
+    LogWarning "Empty ServiceName for package `"$($metadataEntry.Package)`". Skipping."
+    continue
+  }
+  $packagesForToc[$metadataEntry.Package] = $metadataEntry
+}
+
+# Get unique service names and sort alphabetically to act as the service nodes
+# in the ToC
+$services = @{}
+foreach ($package in $packagesForToc.Values) {
+  if ($package.ServiceName -eq 'Other') {
+    # Skip packages under the service category "Other". Those will be handled
+    # later
+    continue
+  }
+  if (!$services.ContainsKey($package.ServiceName)) {
+    $services[$package.ServiceName] = $true
+  }
+}
+$serviceNameList = $services.Keys | Sort-Object
+
+
+$toc = @()
+foreach ($service in $serviceNameList) {
+  Write-Host "Building service: $service"
+
+  $packageItems = @()
+  $packageCount = 0
```

Comment:
This does not appear to be used? Also if it _does_ get used later, wouldn't `$packageItems.Count` be better?